### PR TITLE
[Repair Requests] Prevent out-of-scope create deep-link blank sheet (#303)

### LINKED ISSUE #303: [Repair Requests] Phase 3 - deny and no-open create deep-link for out-of-scope equipment
## Summary
PR-sized batch for repair-request create deep-link behavior.

## Acceptance Criteria
- [ ] Replace the current wrong test expectation before changing hook code
- [ ] `?action=create` still opens a blank sheet
- [ ] `?action=create&equipmentId=X` opens only when `equipment_get` resolves in scope
- [ ] Access denied, not found, or unresolved resolution shows the existing destructive toast
- [ ] Access denied, not found, or unresolved resolution cleans the URL
- [ ] Access denied, not found, or unresolved resolution does not open a blank sheet
- [ ] Existing race-condition coverage remains valid
- [ ] Focused Vitest is observed RED before implementation, then GREEN

## Verification
- `node scripts/npm-run.js run test:run -- 'src/app/(app)/repair-requests/__tests__/useRepairRequestsDeepLink.test.ts'`

## Notes
- Depends on the server deny behavior from #301 and #302

### PR DESCRIPTION
**Pull request by thienchi2109**
## Summary
- Closes #303
- Prevent `?action=create&equipmentId=...` from opening a blank repair request sheet when equipment resolution is denied, missing, invalid, or otherwise unresolved
- Keep plain `?action=create` and valid resolved equipment prefill behavior unchanged
- Split the oversized deep-link test file into focused registrars before changing behavior

## TDD Evidence
- Mechanical split GREEN: `node scripts/npm-run.js run test:run -- 'src/app/(app)/repair-requests/__tests__/useRepairRequestsDeepLink.test.ts'` -> 18 passed\n- RED before hook implementation: same focused suite -> 5 failed / 15 passed on the new no-open expectations\n- Final GREEN: same focused suite -> 20 passed\n\n## Verification\n- `node scripts/npm-run.js run verify:no-explicit-any`\n- `node scripts/npm-run.js run typecheck`\n- `node scripts/npm-run.js run test:run -- 'src/app/(app)/repair-requests/__tests__/useRepairRequestsDeepLink.test.ts'`\n- `node scripts/npm-run.js npx react-doctor@latest . --verbose -y --project nextn --offline --diff main` -> 100 / 100, no issues\n\n## Notes\n- No SQL migration or Supabase DDL in this batch; #301/#302 already provide the server-side deny behavior.\n

---

<a href="https://app.devin.ai/review/thienchi2109/qltbyt-nam-phong/pull/310" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

---
## Summary by cubic
Stop opening a blank repair request sheet for `?action=create&equipmentId=...` when the equipment is invalid, missing, or access is denied. Now we also require a targeted equipment fetch before opening (even if it’s already in the list); plain `?action=create` and valid equipment prefill still work. Closes #303.

- **Bug Fixes**
  - Gate deep-link create on targeted `equipment_get`; require this fetch even if the item is cached, ignore stale IDs, and prevent double-opens.
  - On invalid/missing/denied IDs, show a destructive toast (including the server message when available), clean the URL, and do not open the sheet.
  - Validate `equipmentId`; non-numeric values are treated as errors (not blank create).

- **Refactors**
  - Split deep-link tests into `useRepairRequestsDeepLink.race-cases.ts` and `useRepairRequestsDeepLink.retry-cases.ts`.
  - Use `REPAIR_REQUEST_CREATE_ACTION` and reset resolution state when the create intent is cleared.

<sup>Written for commit aad857ef394682dbfa5aebc4b22eb60854c7cd04. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling and validation in repair request creation with clearer error messages when equipment cannot be found or accessed.
  * Enhanced robustness of the create sheet workflow to prevent unexpected behavior during concurrent requests.

* **Tests**
  * Added comprehensive test coverage for race-condition scenarios and edge cases.
  * Added retry scenario tests to ensure proper behavior during equipment resolution retry flows.

### CODE REVIEW

**Comment by thienchi2109** [comment]
@coderabbitai review

File: src/app/(app)/repair-requests/_hooks/useRepairRequestsDeepLink.ts [hunk 1]
```diff
@@ -156,23 +161,30 @@ export function useRepairRequestsDeepLink(
...
       // Equipment already in list — mark as resolved for create-intent gating
       if (hasCreateAction) {
-        setResolution({ equipmentId: idNum, phase: 'resolved', equipment: existing })
+        setResolution({
+          equipmentId: idNum,
+          phase: 'resolved',
+          equipment: existing,
+          failureMessage: null,
+        })
       }
```
[the remark this answers is not in the document]
Comment (thienchi2109) [reply src/app/(app)/repair-requests/_hooks/useRepairRequestsDeepLink.ts]: Fixed in `88764bf`: create-intent deep links no longer resolve from `allEquipment`; the cache shortcut is now non-create only. Added regression coverage: `requires equipment_get even when targeted equipment is already cached from the list`, observed RED before the hook change and GREEN after.

Verification after the fix:
- `node scripts/npm-run.js run verify:no-explicit-any`
- `node scripts/npm-run.js run typecheck`
- `node scripts/npm-run.js run test:run -- 'src/app/(app)/repair-requests/__tests__/useRepairRequestsDeepLink.test.ts'` -> 21 passed\n- `node scripts/npm-run.js npx react-doctor@latest . --verbose -y --project nextn --offline --diff main` -> 100/100

File: src/app/(app)/repair-requests/_hooks/useRepairRequestsDeepLink.ts [hunk 2]
```diff
@@ -156,23 +161,30 @@ export function useRepairRequestsDeepLink(
...
     ) return
+    if (hasCreateAction) {
+      activeCreateEquipmentIdRef.current = idNum
+    }
     const existing = allEquipment.find(eq => eq.id === idNum)
     if (existing) {
       // Equipment already in list — mark as resolved for create-intent gating
       if (hasCreateAction) {
-        setResolution({ equipmentId: idNum, phase: 'resolved', equipment: existing })
+        setResolution({
```
[the remark this answers is not in the document]
Comment (thienchi2109) [reply src/app/(app)/repair-requests/_hooks/useRepairRequestsDeepLink.ts]: Fixed in `88764bf`: `action=create&equipmentId=...` now falls through to targeted `equipment_get` even when the item is already present in the list cache. Added the cached-list regression test and reran the required gates successfully.
